components/elm/src/main/elm_driver.F90

```diff
@@ -344,41 +339,63 @@ subroutine elm_drv(doalb, nextsw_cday, declinp1, declin, rstwr, nlend, rdate)
              call c14_grc_cf%ZeroDWT(bounds_clump)
              call c14_col_cf%ZeroDWT(bounds_clump)
           end if
-
           call veg_ns%ZeroDWT(bounds_clump)
+          call veg_ps%ZeroDWT(bounds_clump)
+          call t_stopf('cnpvegzero')
+       end if
 
+       if (use_cn .or. use_fates) then
+          call t_startf('cnpzero')
+          call grc_cf%ZeroDWT(bounds_clump)
+          call col_cf%ZeroDWT(bounds_clump)
           call grc_nf%ZeroDWT(bounds_clump)
           call col_nf%ZeroDWT(bounds_clump)
-
-          call veg_ps%ZeroDWT(bounds_clump)
-
           call grc_pf%ZeroDWT(bounds_clump)
           call col_pf%ZeroDWT(bounds_clump)
+          call t_stopf('cnpzero')
+       end if
 
+       call t_startf('cnpvegsumm')
+       if(use_cn) then
```

Comment:
Should this check be `if (use_cn .and. .not. use_fates)`?

Reply:
The two should be mutually exclusive.  you cannot have use_cn = true and use_fates = true.  So when use_cn = True, .not.use_fates is implied.

Reply:
Okay. Sounds good.